main.py

```diff
@@ -575,6 +575,35 @@ def delete_tag_range(self, request: Request) -> JSONResponse:
             raise HTTPException(400, detail=detail)
         return JSONResponse("Operation Successful", status_code=200)
 
+    @rest('v3/interfaces/tag_ranges', methods=['GET'])
+    @validate_openapi(spec)
+    def get_all_tag_ranges(self, _: Request) -> JSONResponse:
+        """Get all tag_ranges and available_tags from interfaces"""
+        result = defaultdict(dict)
+        for switch_id, switch in self.controller.switches.copy().items():
+            for intf_id, interface in switch.interfaces.copy().items():
+                result[switch_id][intf_id] = {
```

Comment:
On second thought, most of the APIs in `topology` or `of_lldp` who are dealing with interface id, is directly using it as keys. Instead of `[dpid][port_number]`, could you simplify one nested level of the response and follow the existing pattern and use `result[interface.id]` instead since `interface.id` is already unique anyway? Let me know if you see a cleaner option than this. 

And then if you need to also filter by switch(es) you can provide a url arg list filtering by dpids in the future, but still only keeping a one level nested dict.

Reply:
Changed